### PR TITLE
Add documentation for dashboards assistant plugin

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
Documentation for dashboards assistant plugin.

The documentation is created by @xluo-aws @wanglam @gaobinlong @raintygao @ruanyl

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6369

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kgcreative** [comment]
This should be "OpenSearch Assistant Toolkit for OpenSearch Dashboards" (if we are specifically referring to the OpenSearch Dashboards features, as opposed to "OpenSearch Assistant Toolkit" for all the backend tooling)

File: _dashboards-assistant/index.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+Each response contains a thumb up and thumb down icon button, you can click on the button to feedback whether you are satisfied with the response.
+
+<img width="600px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/rate.png" alt="Rate the response">
+
+### See how the response was generated
+Under the available suggestions, there will be a link “How was this generated?”, click this link will show which tools are triggered to generate this response. It can be used for trouble shooting. There could be multiple steps if multiple tools are involved. Each step will display tool name with input and output of the tool.
+
+<img width="400px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/traces.png" alt="How was the response generated">
+
+### Continue with a past conversation
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:74]:
```suggestion
### Resuming previous conversations

```

File: _dashboards-assistant/index.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+#### View past conversation
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:75]:
```suggestion

```

File: _dashboards-assistant/index.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+Click the clock icon to open the conversation history panel. The clock icon will become active status and conversations will be displayed clicked after clicked.
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:76]:
```suggestion
To view a previous conversation, select the clock icon to open the conversation history panel and display the chat history. A snapshot of a conversation history is shown in the following image.
```

File: _dashboards-assistant/index.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/conversation-history-entry.png" alt="Show conversation history">
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:78]:
```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/conversation-history-entry.png" alt="Snapshot of conversation history">
```

File: _dashboards-assistant/index.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+### Rate the response
+Each response contains a thumb up and thumb down icon button, you can click on the button to feedback whether you are satisfied with the response.
+
+<img width="600px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/rate.png" alt="Rate the response">
+
+### See how the response was generated
+Under the available suggestions, there will be a link “How was this generated?”, click this link will show which tools are triggered to generate this response. It can be used for trouble shooting. There could be multiple steps if multiple tools are involved. Each step will display tool name with input and output of the tool.
+
+<img width="400px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/traces.png" alt="How was the response generated">
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:72]:
```suggestion
<img width="400px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/traces.png" alt="Snapshot of response generation details">
```

File: _dashboards-assistant/index.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+### Continue with a past conversation
+#### View past conversation
+Click the clock icon to open the conversation history panel. The clock icon will become active status and conversations will be displayed clicked after clicked.
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/conversation-history-entry.png" alt="Show conversation history">
+
+Search conversations by title:
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:80]:
```suggestion
The conversation history can be searched by title, as shown in the following snapshot. 
```

File: _dashboards-assistant/index.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/conversation-history-list.png" alt="List of conversations">
+
+#### Manage past conversation
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:84]:
```suggestion
#### Editing and deleting previous conversations

```

File: _dashboards-assistant/index.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+Click the pencil icon to edit conversation name, type the new conversation name in the below input, then click “Confirm name” button. The new conversation name will be saved.
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:85]:
```suggestion
Select the pencil icon to edit a conversation name and rename it. Select the **Confirm name** button to save the new name. A snapshot of the edit interface is shown in the following image.
```

File: _dashboards-assistant/index.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/edit-conversation-title.png" alt="Edit the conversation title">
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:87]:
```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/edit-conversation-title.png" alt="Editing a conversation name">
```

File: _dashboards-assistant/index.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+Click the trash icon to delete conversation, the delete confirmation modal will be displayed after clicked. Click the “Delete conversation” button to confirm the conversation delete operation. The conversation will be deleted after “Delete conversation” button clicked.
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:89]:
```suggestion
Select the trash icon to delete conversation. The confirmation dialog will appear. Select “Delete conversation”. The conversation is now deleted from your chat history. A snapshot of the delete interface is shown in the following image.
```

File: _dashboards-assistant/index.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/delete-conversation.png" alt="Delete the conversation">
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:91]:
```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/delete-conversation.png" alt="Deleting a conversation">
```

File: _dashboards-assistant/index.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+
+### Share a conversation via notebook
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:93]:
```suggestion
### Sharing a conversation through Notebooks

```

File: _dashboards-assistant/index.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+Click `Save to notebook` button to save your conversation to notebook. All messages between you and LLM model will be saved to observability notebook after entering name for notebook and confirm.
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:94]:
```suggestion
You can use [Notebooks]({{site.url}}{{site.baseurl}}/observing-your-data/notebooks/) to save your conversations. To use this option, select `Save to notebook` from the dropdown menu on the right-side of **OpenSearch Assistant**. Enter a name for the notebook and save it. All conversations (questions and responses) between you and the large language model (LLM) will be saved to this notebook. The following snapshot shows a save notebook interface. 
```

File: _dashboards-assistant/index.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+
+<img width="400px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/save-conversation-to-notebook.png" alt="Save a conversation to notebook">
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:96]:
```suggestion
<img width="400px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/save-conversation-to-notebook.png" alt="Saving a conversation through Notebooks">
```

File: _dashboards-assistant/index.md [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+
+Now you can find your conversation in observability notebook lists and others can see it as well.
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:98]:
```suggestion
You can open your saved notebook or view a list of other notebooks by going to **Observability** > **Notebooks**. The following snapshot shows the the Notebooks interface in OpenSearch Dashboards.
```

File: _dashboards-assistant/index.md [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/conversation-in-notebook.png" alt="Conversation saved to notebook">
```
Comment (vagimeli) [inline _dashboards-assistant/index.md:100]:
```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/conversation-in-notebook.png" alt="List of conversations saved in Notebooks">
```

**Review by vagimeli — COMMENTED** [review]
Please review the doc reviewer's edits and comments. Please make sure the edits have retained technical accuracy. Once the dev/author approves these changes, I'll format the file and replace any low-resolution screenshots. Please provide the testing environment. Following this, I'll schedule the doc for editorial review. Please be available to answer any technical questions. Thank you, Melissa

File: _dashboards-assistant/index.md (hunk 16 above)
Comment (vagimeli) [reply _dashboards-assistant/index.md:100]: Add new section 

## Relate information

The following resources provide more information about the OpenSearch Assistant Toolkit and related components:

- [OpenSearch adds a new generative AI assistant toolkit](https://opensearch.org/blog/opensearch-adds-new-generative-ai-assistant-toolkit/)
- <other resources>

**Comment by vagimeli** [comment]
> This should be "OpenSearch Assistant Toolkit for OpenSearch Dashboards" (if we are specifically referring to the OpenSearch Dashboards features, as opposed to "OpenSearch Assistant Toolkit" for all the backend tooling)

@kgcreative I agree. I've changed doc title.

**Comment by vagimeli** [comment]
> @xluo-aws @wanglam @gaobinlong @raintygao @ruanyl

To move this documentation forward, please provide your technical review and approval. Once I have your approval of my rewrites and edits, I can move the content into editorial review and then merge the PR. Please reach out to me with any questions. Thank you.

File: _dashboards-assistant/index.md [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+redirect_from:
+  - /dashboards-assistant/index/
+---
+
+This is an experimental feature and is not recommended for use in a production environment. For updates on the progress of the feature or if you want to leave feedback, please checkout [GitHub repo](https://github.com/opensearch-project/dashboards-assistant).
+{: .warning}
+
+# Dashboards assistant
+
+The OpenSearch Assistant Dashboards plugin lets you have an OpenSearch assistant to help dashboards users use OpenSearch. With dashboards assistant, user can quickly gain insights on their operational data just with nature language without the need to specialized in query tools and skills.
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _dashboards-assistant/index.md:18]: Note: I'll add the missing link.

[the remark this answers is not in the document]
Comment (lezzago) [reply _dashboards-assistant/index.md:18]: Do we want to talk about the plugin in particular here? The plugin piece is part of the OpenSearch Assistant toolkit to create an chat interface for a LLM-powered OpenSearch Assistant to help users interact with OpenSearch.

**Comment by ylwu-amzn** [comment]
@ruanyl looks some screenshot should be updated

**Comment by raintygao** [comment]
> Will update the doc by next Monday if that's ok. All related team members are taking holidays off this week.

@xluo-aws  According to [release issue](https://github.com/opensearch-project/opensearch-build/issues/4115), pre release is on the end of February 19th. If there are any updates needed, we can complete them before end of 18th, not sure if it's OK.

**Comment by raintygao** [comment]
> @ruanyl looks some screenshot should be updated

@ylwu-amzn  Could you elaborate more?

**Comment by hdhalter** [comment]
> > Will update the doc by next Monday if that's ok. All related team members are taking holidays off this week.
> 
> @xluo-aws According to [release issue](https://github.com/opensearch-project/opensearch-build/issues/4115), pre release is on the end of February 19th. If there are any updates needed we can complete them before end of 18th, not sure if it's OK.

Hi @xluo-aws , tldr: the 18th would be too late to receive updates. Here's why: after tech review is complete (development signs off on the content from a technical perspective), we do a 'doc review' which reviews the content for clarity, format, readability, grammar, etc., and then we do an 'editorial review' which is a final check for any remaining issues (and needs a 2-day window before the release). Ideally, the content should pass the tech review no later than a week before the final date so we have time for the doc and editorial reviews. This is especially important for large PRs like this. 

Thanks to the devs that stepped in to review this!

File: _dashboards-assistant/index.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+Then configure the root `agent_id` through the following API:
+```
+PUT .plugins-ml-config/_doc/os_chat
+{
+    "type":"os_chat_root_agent",
+    "configuration":{
+        "agent_id": "your root agent id"
+    }
+}
+```
```
Comment (lezzago) [inline _dashboards-assistant/index.md:42]: We should call out this is a system index, so in security enabled domains, the user must be super admin to do this.
They can use this guide to make super admin calls: https://opensearch.org/docs/latest/security/configuration/system-indices/

**Comment by vagimeli** [comment]
> > > Will update the doc by next Monday if that's ok. All related team members are taking holidays off this week.
> > 
> > 
> > @xluo-aws According to [release issue](https://github.com/opensearch-project/opensearch-build/issues/4115), pre release is on the end of February 19th. If there are any updates needed we can complete them before end of 18th, not sure if it's OK.
> 
> Hi @xluo-aws , tldr: the 18th would be too late to receive updates. Here's why: after tech review is complete (development signs off on the content from a technical perspective), we do a 'doc review' which reviews the content for clarity, format, readability, grammar, etc., and then we do an 'editorial review' which is a final check for any remaining issues (and needs a 2-day window before the release). Ideally, the content should pass the tech review no later than a week before the final date so we have time for the doc and editorial reviews. This is especially important for large PRs like this.
> 
> Thanks to the devs that stepped in to review this!

@hdhalter @xluo-aws I've moved this into editorial review. I'll address editor comments when they come in. If any content is added after the editorial review, I'll review them and ask the editor to review only the newly added sections. This will help keep us on track for publishing by release date.

File: _dashboards-assistant/index.md [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/edit-conversation-title.png" alt="Editing a conversation name">
+
+Select the trash icon to delete conversation. The confirmation dialog will appear. Select “Delete conversation”. The conversation is now deleted from your chat history. A snapshot of the delete interface is shown in the following image.
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/delete-conversation.png" alt="Deleting a conversation">
+
+### Sharing a conversation through Notebooks
+
+You can use [Notebooks]({{site.url}}{{site.baseurl}}/observing-your-data/notebooks/) to save your conversations. To use this option, select `Save to notebook` from the dropdown menu on the right-side of **OpenSearch Assistant**. Enter a name for the notebook and save it. All conversations (questions and responses) between you and the large language model (LLM) will be saved to this notebook. The following snapshot shows a save notebook interface.
```
Comment (natebower) [inline _dashboards-assistant/index.md]:
```suggestion
You can use [Notebooks]({{site.url}}{{site.baseurl}}/observing-your-data/notebooks/) to save your conversations. To use this option, select `Save to notebook` from the dropdown menu to the right of **OpenSearch Assistant**. Enter a name for the notebook and save it. All conversations (questions and responses) between you and the large language model (LLM) will be saved to this notebook. The following snapshot shows a save notebook interface. 
```

Comment (natebower) [inline _dashboards-assistant/index.md]: "screenshot" instead of "snapshot"?

File: _dashboards-assistant/index.md [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+
+<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/delete-conversation.png" alt="Deleting a conversation">
+
+### Sharing a conversation through Notebooks
+
+You can use [Notebooks]({{site.url}}{{site.baseurl}}/observing-your-data/notebooks/) to save your conversations. To use this option, select `Save to notebook` from the dropdown menu on the right-side of **OpenSearch Assistant**. Enter a name for the notebook and save it. All conversations (questions and responses) between you and the large language model (LLM) will be saved to this notebook. The following snapshot shows a save notebook interface. 
+
+<img width="400px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/save-conversation-to-notebook.png" alt="Saving a conversation through Notebooks">
+
+You can open your saved notebook or view a list of other notebooks by going to **Observability** > **Notebooks**. The following snapshot shows the the Notebooks interface in OpenSearch Dashboards.
```
Comment (natebower) [inline _dashboards-assistant/index.md]:
```suggestion
You can open your saved notebook or view a list of other notebooks by going to **Observability** > **Notebooks**. The following snapshot shows the Notebooks interface in OpenSearch Dashboards.
```

Comment (natebower) [inline _dashboards-assistant/index.md]: "screenshot" instead of "snapshot"?

**Review by natebower — COMMENTED** [review]
@vagimeli @ruanyl Please see my comments and changes and let me know if you have any questions. Thanks!

File: _dashboards-assistant/index.md [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+## Configuring the plugin
+Go to the [Flow Framework plugin](https://github.com/opensearch-project/flow-framework) documentation to get the default chatbot template. You can modify this template to use your own model and customize the chatbot tools. Go to the [Getting started guide](https://github.com/opensearch-project/dashboards-assistant/blob/main/GETTING_STARTED_GUIDE.md) for step-by-step instructions. 
+
+## Using OpenSearch Assistant in OpenSearch Dashboards
+The following sections guide you through using OpenSearch Assistant in OpenSearch Dashboards.
+
+### Start the conversation
+
+
+Start the conversation by entering a question in the **Ask a question** search box or using the shortcut `ctrl + /`. An interface snapshot is shown in the following image.
```
[the remark this answers is not in the document]
Comment (vagimeli) [reply _dashboards-assistant/index.md]: Yes, I'll make changes throughout.

**Comment by vagimeli** [comment]
> @ruanyl looks some screenshot should be updated

@ruanyl If you send me the test environment, I can update all screenshots.

File: _dashboards-assistant/index.md (hunk 18 above)
Comment (vagimeli) [reply _dashboards-assistant/index.md:44]: @lezzago Please review the following callout text to make sure I've captured your comment accurately:

_This example is a system index. In security-enabled domains, only super administrators have access to execute this code. For details about making super admin calls, see the [System indexes](https://opensearch.org/docs/latest/security/configuration/system-indices/) guide. For access permission, contact your IT administrator._

**Comment by vagimeli** [comment]
> @vagimeli @ruanyl Please see my comments and changes and let me know if you have any questions. Thanks!

@natebower @ruanyl I've accepted the editorial review comments.

**Comment by ylwu-amzn** [comment]
> > @ruanyl looks some screenshot should be updated
> 
> @ylwu-amzn Could you elaborate more?

Shared you a latest test cluster. You can take new screenshots there.

**Comment by vagimeli** [comment]
> > > @ruanyl looks some screenshot should be updated
> > 
> > 
> > @ylwu-amzn Could you elaborate more?
> 
> Shared you a latest test cluster. You can take new screenshots there.

@ylwu-amzn @ruanyl I've replaced screenshots based on the the test environment you shared. Please provide your approval of this PR if its ready for merging. Thank you, Melissa

File: _dashboards/dashboards-assistant/index.md [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+{
+    "type":"os_chat_root_agent",
+    "configuration":{
+        "agent_id": "your root agent id"
+    }
+}
+```
+{% include copy-curl.html %}
+
+This example is a system index. In security-enabled domains, only super administrators have access to execute this code. For details about making super admin calls, see the [System indexes]({{site.url}}{{site.baseurl}}/security/configuration/system-indices/) guide. For access permission, contact your IT administrator.
```
Comment (vagimeli) [inline _dashboards/dashboards-assistant/index.md:43]: @natebower Please edit lines 43-44.

**Review by vagimeli — APPROVED** [review]
Doc and editorial reviews completed. Editorial feedback addressed.

**Pull request merged**